src/Microsoft.OData.Core/ErrorUtils.cs

```diff
@@ -43,8 +43,12 @@ internal static void GetErrorDetails(ODataError error, out string code, out stri
         /// <param name="writer">The Xml writer to write to.</param>
         /// <param name="error">The error instance to write.</param>
         /// <param name="includeDebugInformation">A flag indicating whether error details should be written (in debug mode only) or not.</param>
-        /// <param name="maxInnerErrorDepth">The maximum number of nested inner errors to allow.</param>
-        internal static void WriteXmlError(XmlWriter writer, ODataError error, bool includeDebugInformation, int maxInnerErrorDepth)
+        /// <param name="messageWriterSettings">Configuration settings of the OData writer.</param>
+        internal static void WriteXmlError(
+            XmlWriter writer,
+            ODataError error,
+            bool includeDebugInformation,
+            ODataMessageWriterSettings messageWriterSettings)

```

Comment:
Is there any advantage of passing the whole `ODataMessageWriterSettings` here in place of the int? Do these methods require other properties from `ODatamessageWriterSettings` other than the max nesting depth?

Reply:
@ElizabethOkerio Yes. We're using the same `ODataMessageWriterSettings` object to check whether the backward compatibility flag `UseLegacyODataInnerErrorSerialization` flag has been toggled. I reasoned it's better to pass the object since it contains settings instead of adding an extra parameter. That way we don't need to change the method definition if another writer settings needs to be referenced.